Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTargetDescriptorFactory.cpp

```diff
@@ -108,7 +114,21 @@ namespace TestImpact
         const auto& target = buildTarget[Keys[TargetKey]];
         descriptor.m_name = target[Keys[NameKey]].GetString();
         descriptor.m_outputName = target[Keys[OutputNameKey]].GetString();
-        descriptor.m_path = target["path"].GetString();
+        descriptor.m_path = target[Keys[PathKey]].GetString();
+
+        if (const auto targetTypeString = target[Keys[TargetTypeKey]].GetString();
+            strcmp(targetTypeString, Keys[ProductionTargetTypeKey]) == 0)
+        {
+            descriptor.m_type = TargetType::ProductionTarget;
+        }
+        else if (strcmp(targetTypeString, Keys[TestTargetTypeKey]) == 0)
+        {
+            descriptor.m_type = TargetType::TestTarget;
+        }
+        else
+        {
+            throw(ArtifactException(AZStd::string::format("Unexpected target type '%s'", targetTypeString)));
+        }
```

Comment:
Don't use `strcmp`. `rapidjson::GenericValue` includes an [`operator==` overload for comparing to a C string](https://github.com/Tencent/rapidjson/blob/64faab2e9235ed51f37ef592010b35fc7f7324ec/include/rapidjson/document.h#L1067).
```suggestion
        if (const auto& targetType = target[Keys[TargetTypeKey]];
            targetType == Keys[ProductionTargetTypeKey])
        {
            descriptor.m_type = TargetType::ProductionTarget;
        }
        else if (targetType == Keys[TestTargetTypeKey])
        {
            descriptor.m_type = TargetType::TestTarget;
        }
        else
        {
            throw(ArtifactException(AZStd::string::format("Unexpected target type '%s'", targetType.GetString())));
        }
```

Reply:
V. useful, did not know that! Fixed 👍